File: app/src/main/java/de/tum/in/tumcampusapp/models/efa/StationResult.java
package de.tum.in.tumcampusapp.models.efa;

public class StationResult {
    public String station;
    public String id;
    public int quality;

    @SuppressWarnings("unused")
    public StationResult() {
        // no-args constructor for GSON
    }

    public StationResult(String station, String id, int quality) {
        this.station = station;
        this.id = id;
        this.quality = quality;
    }

    @Override
    public String toString() {
        return station;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/models/efa/StationResult.kt
package de.tum.`in`.tumcampusapp.models.efa

import com.google.gson.Gson
import de.tum.`in`.tumcampusapp.models.dbEntities.Recent

data class StationResult(var station: String = "",
                         var id: String = "",
                         var quality: Int = 0) {

    override fun toString(): String = station

    companion object {
        fun fromRecent(r: Recent): StationResult? {
            return Gson().fromJson(r.name, StationResult::class.java)
        }
    }
}